packages/framework/ai-collab/src/test/explicit-strategy/agentEditing.spec.ts

```diff
@@ -0,0 +1,271 @@
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: Root }));
+		view.initialize({ arr: [1, 2, 3] });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({ __fluid_objectId: "Root1", arr: [1, 2, 3] }),
+		);
+
+		// const hydratedObject = hydrate(Root, new Root({ arr: [1, 2, 3] }));
+		// assert.equal(
+		// 	toDecoratedJson(idGenerator, hydratedObject),
+		// 	JSON.stringify({ __fluid_objectId: "Root0", arr: [1, 2, 3] }),
+		// );
+	});
+});
+
+describe("Makes TS type strings from schema", () => {
+	it("for objects with primitive fields", () => {
+		const testSf = new SchemaFactory("test");
+		class Foo extends testSf.object("Foo", {
+			x: testSf.number,
+			y: testSf.string,
+			z: testSf.optional(testSf.null),
+		}) {}
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(Foo)),
+			"interface Foo { x: number; y: string; z: null | undefined; }",
+		);
+	});
+
+	// This test fails due to the fact that identifier fields are incorrectly set as optional in the JSON Schema
+	it.skip("for objects with identifier fields", () => {
+		const testSf = new SchemaFactory("test");
+		class Foo extends testSf.object("Foo", {
+			y: testSf.identifier,
+		}) {}
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(Foo)),
+			"interface Foo { y: string; }",
+		);
+	});
+
+	it("for objects with polymorphic fields", () => {
+		const testSf = new SchemaFactory("test");
+		class Bar extends testSf.object("Bar", {
+			z: testSf.number,
+		}) {}
+		class Foo extends testSf.object("Foo", {
+			y: demoSf.required([demoSf.number, demoSf.string, Bar]),
+		}) {}
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(Foo)),
+			"interface Foo { y: number | string | Bar; } interface Bar { z: number; }",
+		);
+	});
+
+	it("for objects with array fields", () => {
+		const testSf = new SchemaFactory("test");
+		class Foo extends testSf.object("Foo", {
+			y: demoSf.array(demoSf.number),
+		}) {}
+		const stringified = JSON.stringify(getJsonSchema(Foo));
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(Foo)),
+			"interface Foo { y: number[]; }",
+		);
+	});
+
+	it("for objects with nested array fields", () => {
+		const testSf = new SchemaFactory("test");
+		class Foo extends testSf.object("Foo", {
+			y: demoSf.array([
+				demoSf.number,
+				demoSf.array([demoSf.number, demoSf.array(demoSf.string)]),
+			]),
+		}) {}
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(Foo)),
+			"interface Foo { y: (number | (number | string[])[])[]; }",
+		);
+	});
+
+	it("for objects in the demo schema", () => {
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(RootObject)),
+			"interface RootObject { str: string; vectors: Vector[]; bools: boolean[]; } interface Vector { x: number; y: number; z: number | undefined; }",
+		);
+	});
+});
+
+describe.skip("llmClient", () => {
```

Comment:
Note about why we're skipping it, and reference to tracking item? Or should it not be here?

Reply:
This was a left-over test from the SharedTree team. Looks to cover the tree node -> json schema -> does open AI API accept the response. I'll delete this one but i think it's worth us adding tests in regards to this topic eventually. It has been removed for now in commit bac309434408ff1fd2adfbc0dcacf19c89f40ee8